src/lib/types.hpp

```diff
@@ -84,13 +86,23 @@ using pmr_ring_buffer = boost::circular_buffer<T, PolymorphicAllocator<T>>;
 
 using ChunkOffset = uint32_t;
 
-// Used to represent NULL values
 constexpr ChunkOffset INVALID_CHUNK_OFFSET{std::numeric_limits<ChunkOffset>::max()};
+constexpr ChunkID INVALID_CHUNK_ID{std::numeric_limits<ChunkID::base_type>::max()};
 
 struct RowID {
-  ChunkID chunk_id{0};
+  ChunkID chunk_id{INVALID_CHUNK_ID};
   ChunkOffset chunk_offset{INVALID_CHUNK_OFFSET};
 
+  RowID() = default;
+
+  RowID(const ChunkID chunk_id, const ChunkOffset chunk_offset) : chunk_id(chunk_id), chunk_offset(chunk_offset) {
+    DebugAssert((chunk_offset == INVALID_CHUNK_OFFSET) == (chunk_id == INVALID_CHUNK_ID),
```

Comment:
Okay this took me a while. I’d prefer `DebugAssert(*this != NULL_ROW_ID, "");`

Reply:
Clever

Reply:
but not doing what I intended. What this Asserts is that "If you pass in one of the arguments as INVALID/NULL, the other has to be INVALID/NULL as well. This makes sure there is just one value representing an invalid row id."

Reply:
Should it ever be possible to use `INVALID_CHUNK_OFFSET` and `INVALID_CHUNK_OFFSET` separately?

Reply:
Don't know what you are getting at. Maybe? Why forbid it?